Observer/Frontend/CustomerLoggedIn.php

```diff
@@ -70,8 +70,21 @@ public function execute(Observer $observer)
             $this->sailthruClient->logger($e);
         }
     }
-    private function shouldUpdateSubscriptionStatus($newsletterList, $userData)
+
+    /**
+     * Should update subscription status
+     *
+     * @param $newsletterList
+     * @param $userData
+     *
+     * @return bool
+     */
+    protected function shouldUpdateSubscriptionStatus($newsletterList, $userData)
     {
+        if (!isset($userData['optout_email']) || empty($newsletterList) || empty($userData['lists'])) {
+            return false;
+        }
+
```

Comment:
@sofvlad also in "Helper/Customer.php" replace methods getObject() and getObjectById() by:
```
    /**
     * @param int|string $param
     *
     * @return MageCustomer
     * @throws \Magento\Framework\Exception\NoSuchEntityException
     */
    public function getObject($param)
    {
        $this->customerModel->setWebsiteId($this->storeManager->getStore()->getWebsiteId());
        try {
            $this->customerModel->loadByEmail($param);
        } catch (\Exception $e) {
            throw new \Exception($e->getTraceAsString());
        }

        return $this->customerModel;
    }

    /**
     * @param $id
     *
     * @return MageCustomer
     * @throws \Magento\Framework\Exception\NoSuchEntityException
     */
    public function getObjectById($id)
    {
        $this->customerModel->setWebsiteId($this->storeManager->getStore()->getWebsiteId());
        try {
            $this->customerModel->load($id);
        } catch (\Exception $e) {
            throw new \Exception($e->getTraceAsString());
        }

        return $this->customerModel;
    }
```